### ISSUE TITLE
Review: Ch 4 (viz)

### ISSUE THREAD

**Issue by leem44** [UBC-DSCI/introduction-to-datascience#104]
# Reviewer E: 
- [x] Section 4.5.4 the formula interface is used with facet_grid(). I believe the preferred approach now is to use the ‘rows’ and ‘cols’ parameters instead. 
- [x] Ch title/subtitles: The actual titles are fine, but for some reason there are unneeded extra levels in the subsection numbers. This is immaterial but unaesthetic

- ~~I really like the introduction. The only thing I thought could be expanded on was the distinction between exploratory and explanatory visualization and how one might approach visualization as part of EDA versus visualizations to document a result~~
- [x] I struggled with the data in the example here. What are the date versus date_decimal fields? 
- [x] When reading in the Manua Loa data, no explanation is given for the filter statement (ppm>0). This was somewhat distracting, so it might be worth either explaining or preprocessing the data to have the correct subset before it is read in. 
- [x] The theme() function is very complex and has many potential arguments. I would recommend either taking the time to explain how it works in more detail or removing it from the text. I worry showing one single example without explanation may lead students to a misunderstanding of what it is.
- ~~In section 4.5.2, the question asked of the plot could easily be answered by a table. Is there a risk here of making students feel that they must plot everything? More broadly, should how to make a good table be covered in parallel as another aspect of data visualization?~~
- [x] Consider showing how to modify an existing ggplot2 object instead of repeatedly recreating (e.g. islands_bar + coord_flip() instead of repeating code to make islands_bar)
- [x] Love the section on explaining a visualization! 
- ~~The section on saving plots seems long and unusually detailed (e.g. raster versus vector images), and I might consider removing. This is all interesting stuff, but it feels very in-depth compared to the higher level treatment of core data analysis and modeling principles. In particular, the book assumes students are working in notebooks where saving images is generally not needed.~~

**Comment by leem44** [comment]
Reviewer B: 
- prefer labelling the subsections presenting examples using the geometric object in question (makes it easier to find code to create specific visualizations, especially since searching bookdown-generated webpages can be difficult). Timbers labels them with the dataset name. Perhaps the subsections could be renamed to
i.	4.5.1 Mauna Loa CO2 data with linegraphs
ii.	4.5.2 Island landmass data with barplots
iii.	4.5.3 Old faithful data with scatterplots
iv.	4.5.4 Michelson speed of light with histograms
- The absence of side-by-side boxplots in the text is a glaring one. Boxplots empower students greatly at they provide them with the means of not just comparing averages between groups, but distributions between groups. That being said, I do appreciate that they are very confusing at first and take a lot of ink to explain well.

**Comment by trevorcampbell** [comment]
# Reviewer D
-  don’t fully agree that a visualization is intended to answer questions – often a visualization is intended to generate questions.  The author should allow for the latter possibility in their explanations.
- I like the emphasis on the iterative process of creating a figure – it is excellent!
- When introducing the different types of plots (e.g., line plots, histograms, scatter plots, bar plots), the authors should show a figure which illustrates all of these types of plots
- When explaining how ggplot should be used to construct a visualization, it would help to use the analogy of creating a painting: we start with a black canvass, we add the graph axes which make it clear what variables we are plotting, we then add visual elements within the region defined by those axes such a bars, points, lines using the concept of geoms and then we annotate the graph with axis labels, titles, etc. 
- When introducing histograms, can the authors discuss the choice of binwidth for geom_histogram()?
- I would love it if the section on Explaining the Visualization would be expanded to its own chapter and elaborated upon more

**Comment by trevorcampbell** [comment]
# Reviewer A
- p79 (instead of distribution / relationship of) I suggest "relationship between"
- p80 The ggthemes package also has a colorblind safe palette, it would be wise to mention that here for ease of use.
- p80 Will a true novice know what overplotting is? It might be worth defining.
- p81 around the code block: I would add some text here for the novice reader.
- p88 Maybe clarify what the values passed to legend.position mean. That will help a novice extend the idea.
- p89 faithful isn't a tibble by default. You can run faithful <- as.tibble(faithful) behind the scenes, this might confuse a novice.
- p93: Talk through the arguments to geom_vline for the reader the first time you use it.
- p102: ypesetting: On this page we have top and bottom, this is likely a sizing issue. figures should be left/right per the caption.
- You mentioned using better color/fill palettes but didn't illustrate their use. I would include that somewhere in the chapter.

# Trevor
- Figure 4.16 on page 102 does not highlight the artifacts the same way the web version does... did we accidentally load the same image twice?

**Comment by ttimbers** [comment]
# Reviewer C
- Section 4.4.0.1: I strongly recommend including “don’t vs do” examples here - show people the bad plot, then show them the plot with that particular problem corrected. Without that, the general rules are difficult to interpret.
- Fig 4.1: worth pointing out that the vertical axis doesn’t start at zero (which meany people regard as misleading)
- Section 4.2: “If there are xlab and ylab arguments, why isn’t there a fontsize argument?” asks the reader.
- Section 4.5.4: You might want to explain this for people who don’t have a physics background…
- Section 4.5.4:  I think it’s a bit of a stretch to introduce the tilde operator in passing like ths… I also think many people will read it as a dash.
- Our in-text comment on Fig 4.14: I would leave it up to the reader to decide if it’s impressive or not.
- Note in section 4.7.0.1: …or because there’s an embedded raster image that’s very large and is being dynamically rescaled… Again, if this fact isn’t used elsewhere in the lesson, can it be removed to reduce learning load?
- In response to how we get the file info out of the saved plots: You have not introduced file_info, nor have you shown how to use [“label”] to pull something out of a named list (which is what I presume this is). Again, I think a consolidation pass is necessary to make sure there aren’t surprises like this.
- Last paragraph of the chapter: I’m pretty sure many readers will need the difference pointed out - there’s a lot going on here, an the fact that the file sizes are different can easily be lost.

**Comment by leem44** [comment]
Adding Tiffany's comment from #92 
We should really introduce colour palettes in chapter 4 in the textbook (to make this book more standalone, we currently only do this in the worksheet/tutorial), and then just use it here and not explain it (since the readers will then know it)

**Comment by ttimbers** [comment]
From Reviewer E

> The section on saving plots seems long and unusually detailed (e.g. raster versus vector images), and I might consider removing. This is all interesting stuff, but it feels very in-depth compared to the higher level treatment of core data analysis and modeling principles. In particular, the book assumes students are working in notebooks where saving images is generally not needed.

I disagree with this comment and feedback and think we should keep this section. Will not address.

**Comment by ttimbers** [comment]
From Reviewer B

> The absence of side-by-side boxplots in the text is a glaring one. Boxplots empower students greatly at they provide them with the means of not just comparing averages between groups, but distributions between groups. That being said, I do appreciate that they are very confusing at first and take a lot of ink to explain well.

I think boxplots are most useful when comparing many distributions - so many that you cannot plot the actual distributions in something like a jitter/strip plot or ridge plot. Also, we don't get to statistical methods that compare distributions. For both these reasons I believe this is out of scope for our book. Will not address.

**Comment by ttimbers** [comment]
From Reviewer D

> I would love it if the section on Explaining the Visualization would be expanded to its own chapter and elaborated upon more

I think this is a good idea but beyond the scope of this revision. Will add to the version 2 issue as a possible future improvement.

**Comment by ttimbers** [comment]
From Reviewer E:

> The theme() function is very complex and has many potential arguments. I would recommend either taking the time to explain how it works in more detail or removing it from the text. I worry showing one single example without explanation may lead students to a misunderstanding of what it is.

I will partially address by stating that the theme() function is very complex and has many potential arguments. And pointing interested readers to docs for this.

**Comment by ttimbers** [comment]
From Reviewer C:

> Section 4.4.0.1: I strongly recommend including “don’t vs do” examples here - show people the bad plot, then show them the plot with that particular problem corrected. Without that, the general rules are difficult to interpret.

I feel like we do that for each of the examples - we start out with the defaults and then iterate to improve it. I get what they are saying about common mistakes people make, but I think that we have 1 viz chapter in a book, not a book on viz, so there is only so much we can do!

**Comment by ttimbers** [comment]
@trevorcampbell - Re: "Figure 4.16 on page 102 does not highlight the artifacts the same way the web version does... did we accidentally load the same image twice?"

No, it does not in the PDF - this was something we need to fix. It is not simple nor straightforward due the the svg. Will move this to the formatting pass.

**Comment by ttimbers** [comment]
From Reviewer E:

>In section 4.5.2, the question asked of the plot could easily be answered by a table. Is there a risk here of making students feel that they must plot everything? More broadly, should how to make a good table be covered in parallel as another aspect of data visualization?

I agree tables are useful, but visualizations are usually more effective at communication. For example, the landmasses bar plots are immediately easier to to see the biggest 7, and see how much bigger they are. This is definitely not as easy, or immediately apparent with the same data presented as a table. Rarely, do I think tables are more effective than visualizations... They are sometimes convenient, but rarely more effective in my opinion. I guess, they are more useful for precisely communicating values...

I think putting an emphasis on visualization in our book, over tables is the right move. A focus on tables is suitable in a DSCI comm course book I think.

**Comment by ttimbers** [comment]
From Reviewer C:

> Fig 4.1: worth pointing out that the vertical axis doesn’t start at zero (which meany people regard as misleading)

My issue is that this is pretty nuanced (or advanced) for our audience. It's a complicated thing to explain and discuss. This is something they would learn in the DSCI minor third year course on data viz, not in DSCI 100 - the audience for which this book was written. Thus I think I am not going to point it out.

**Comment by ttimbers** [comment]
From Reviewers E & D:

>  I really like the introduction. The only thing I thought could be expanded on was the distinction between exploratory and explanatory visualization and how one might approach visualization as part of EDA versus visualizations to document a result
>
> don’t fully agree that a visualization is intended to answer questions – often a visualization is intended to generate questions. The author should allow for the latter possibility in their explanations.

I do see where they are coming from. Viz is used in two contexts usually - exploratory data analysis (EDA) for the analyst during their analysis, and then data viz for communication to stakeholders/readers/etc. But we don't get into this detail of how the workflows differ here. Again, this is one chapter in an intro DS course, not a course on Data Viz.

Additionally, even if you are doing EDA you have exploratory Q's. So I am thinking of not addressing these two points and leaving the intro as it is.

**Comment by ttimbers** [comment]
Closed by #263
